File: Design_Pattern_and_Principles/DependencyInjectionExample/CustomerService.java
public class CustomerService {
    private CustomerRepository repository;
    public CustomerService(CustomerRepository repository) {
        this.repository = repository;
    }

    public void lookupCustomer(String id) {
        String data = repository.findCustomerById(id);
        System.out.println("[Service Logging Logged Request Output]: " + data);
    }
}

File: Week_1/Design_Pattern_and_Principles/DependencyInjectionExample/CustomerService.cs
using System;

public class CustomerService {
    private ICustomerRepository repository;

    public CustomerService(ICustomerRepository repository) {
        this.repository = repository;
    }

    public void LookupCustomer(string id) {
        string data = repository.FindCustomerById(id);
        Console.WriteLine("[Service Logging Logged Request Output]: " + data);
    }
}
